src/python/pants/option/global_options.py

```diff
@@ -273,6 +273,16 @@ def register_bootstrap_options(cls, register):
       help='Create a new pantsd server, and use it, and shut it down immediately after. '
            'If pantsd is already running, it will shut it down and spawn a new instance (Beta)')
 
+    # NB: We really don't want this option to invalidate the daemon, because different clients might have
+    # different needs. For instance, an IDE might have a very long timeout because it only wants to refresh
+    # a project in the background, while a user might want a shorter timeout for interactivity.
```

Comment:
This note could be made a bit more clear what the takeaway is. Maybe something like:

> NB: We intentionally set the default this high because a timeout will raise an exception that then invalidates the daemon. Still, we set this as an option because different clients have different needs, such as an IDE using a very long timeout because it only wants to refresh a project in the background vs. a user wanting a shorter timeout for interactivity.

However, question on this. Does it make sense for the examples to be based on different "clients," i.e. runtime contexts like IDE vs interactive user? I think the relevant context is how long the pants _goals_ take to run, e.g. `./pants list` might be really fast but `./pants test` might take a long time. Whether you run Pants through your IDE or on the command line, it doesn't really matter; what matters is the goals you use.

Reply:
So, a couple of points: This option doesn't invalidate the daemon, especially because it would be really annoying if the IDE is running something in the background, and a client makes a request with a different timeout and crashes the run of the IDE (or vice versa).

Then, the timeout should be set according to "what I expect the other runs are doing", not "how long this particular run is going to take". Since in general this is unknowable, I'm not sure how to phase it better. That said, I agree that 60s might be too long, I'm more than open to suggestions for other numbers.

Reply:
> Then, the timeout should be set according to "what I expect the other runs are doing", not "how long this particular run is going to take". Since in general this is unknowable, I'm not sure how to phase it better. That said, I agree that 60s might be too long, I'm more than open to suggestions for other numbers.

Our p90 is way past 60 seconds, so if anything, 60 isn't long enough. Something like 10 minutes might be better, but is easy to iterate on and/or override in our repo.